### PR TITLE
8344365: SecurityManager cleanups in java.sql and java.sql.rowset modules

### PR DESCRIPTION
**Pull request by eirbjo**
Please review this PR which cleans up SecurityManager-related code in `java.sql` and `java.sql.rowset` modules post JEP-486

There are quite a few changes to review, but all relatively straightforward:

`DriverManager`
* Remove `SecurityManager::checkPermission` calls in the `setLogWriter`, `setLogStream` and `deregisterDriver` methods
* Remove two now-unused package private SQLPermission constants
* `ensureDriversInitialized` is updated to remove `AccessController::doPrivileged` when reading a system property and when initializing drivers

`CachedRowSetImpl`
*  Remove `AccessController::doPrivileged` when getting a `SyncFactory` instance
* `getObject` is update to remove a call to `ReflectUtil::checkPackageAccess`

`CachedRowSetWriter`
* A call to `ReflectUtil::checkPackageAccess` is removed.

`SerialJavaObject`
*  `getFields` is updated to remove call to `ReflectUtil::checkPackageAccess`. `@CallerSensitive` is no longer needed for this method.
* The test `CheckCSMs.java` is updated to remove references to `SerialJavaObject:getFields`

`SyncFactory`
* `initMapIfNecessary` is updated to remove call to `AccessController::doPrivileged` when reading system properties and when reading properties from an input stream
* `getInstance` is updated to remove calls to `ReflectUtil::checkPackageAccess`
* `setLogger` method is updated to remove call to `SecurityManager::checkPermission`
* `setJNDIContext` methods are updated to remove call to `SecurityManager::checkPermission`

`RowsetProvider`
*  Static initializer is updated to call `System::getProperty` directly
* `newFactory` is updated to call `System::getProperty` directly
* `newFactory` is updated to not call `ReflectUtil.checkPackageAccess`
* `getContextClassLoader` is updated to not call `AccessController::doPrivileged`
* `getFactoryClass` is updated to not call `ReflectUtil.checkPackageAccess`
* `getSystemProperty` is removed

`SQLInputImpl`
*  A call to `ReflectUtil::checkPackageAccess` is removed

 `TestPolicy.java` in `test/java/sql/testng/util`
* This  is now unused and removed

Ran `test/jdk/java/sql` and `test/jdk/javax/sql` tests locally. GHA results pending.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8344365](https://bugs.openjdk.org/browse/JDK-8344365): SecurityManager cleanups in java.sql and java.sql.rowset modules (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Brent Christian](https://openjdk.org/census#bchristi) (@bchristi-git - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22185/head:pull/22185` \
`$ git checkout pull/22185`

Update a local copy of the PR: \
`$ git checkout pull/22185` \
`$ git pull https://git.openjdk.org/jdk.git pull/22185/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22185`

View PR using the GUI difftool: \
`$ git pr show -t 22185`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22185.diff">https://git.openjdk.org/jdk/pull/22185.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22185#issuecomment-2481500647)
</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
I think Brent and/or Lance have been working on this already. If you are taking this, can you remove src/java.sql.rowset/share/classes/com/sun/rowset/internal/XmlReaderContentHandler.java from the patch as it introduces a behavioural change that may require further work.

**Comment by eirbjo** [comment]
> I think Brent and/or Lance have been working on this already.

Happy to yield if Brent/Lance has something cooking. Otherwise, perhaps their time is better spent as reviewers.

> If you are taking this, can you remove src/java.sql.rowset/share/classes/com/sun/rowset/internal/XmlReaderContentHandler.java from the patch as it introduces a behavioural change that may require further work.

Seems like a clean inline, how does this introduce a behavioural change?

`ReflectUtil`:

```
public static Class<?> forName(String name) throws ClassNotFoundException {
    return Class.forName(name);
}
```

**Comment by AlanBateman** [comment]
> Seems like a clean inline, how does this introduce a behavioural change?

Class.forName's behavior depends on the caller's defining class loader. I don't know where Brent and/or Lance is on the changes for this module but if you are taking it then I would prefer if the changes to XmlReaderContentHandler were dropped.

**Comment by eirbjo** [comment]
> Class.forName's behavior depends on the caller's defining class loader. I don't know where Brent and/or Lance is on the changes for this module but if you are taking it then I would prefer if the changes to XmlReaderContentHandler were dropped.

Thanks, makes sense. I have reverted changes in `XmlReaderContentHandler`

**Review by LanceAndersen — COMMENTED** [review]
Thanks for tackling this Eirik.  I made a pass through and I think the changes are reasonable.

I have asked Sean to also make a pass as well

**Review by RogerRiggs — APPROVED** [review]
Looks fine.

**Comment by eirbjo** [comment]
Hey @RogerRiggs, I needed to merge with master to resolve a conflict after the integration of your PR #22041.

Would you mind taking another look at the merge commit 3f1df59 and re-review this PR?

While looking at your change in #22041, I noticed you removed `ObjectStreamField#getType` from `CheckCSM.KNOWN_NON_FINAL_CSMS`, but that it remains in `CheckCSM.UNSUPPORTED_VIRTUAL_METHODS`.

CheckCSM runs fine with this method removed from `UNSUPPORTED_VIRTUAL_METHODS` which leads me to think it may have been a leftover?

If it was, I'm happy to remove it while I'm visiting this code anyhow, or you could clean it up via some other PR.

Thanks :-)

**Comment by RogerRiggs** [comment]
Yes, please cleanup the leftover mention, in CheckCSMs. of ObjectStreamField.getType().
And I'll re-review.

**Review by RogerRiggs — APPROVED** [review]
Looks good, thanks for the update.

**Comment by RogerRiggs** [comment]
Leave the test with an empty set of methods for CallerSensistiveAdepter will make it clearer to check for matching test conditions.

File: test/jdk/jdk/internal/reflect/CallerSensitive/CheckCSMs.java
Comment (liach) [inline test/jdk/jdk/internal/reflect/CallerSensitive/CheckCSMs.java:1]: Your mailing list message recommends removal of this test. This test covers more than ensuring that CSMs are final or static; it actually scans all platform classes to ensure CSMs are present wherever they are needed.

Comment (eirbjo) [reply test/jdk/jdk/internal/reflect/CallerSensitive/CheckCSMs.java:1]: Sure, I'll leave the test with the empty set as suggested by Roger.

**Review by bchristi-git — APPROVED** [review]
I think the changes look fine.

**Comment by bchristi-git** [comment]
I submitted an internal automated test run and it passes cleanly.

**Pull request closed without merging**